apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/SnifferConfigInitializer.java

```diff
@@ -202,6 +210,36 @@ private static void overrideConfigBySystemProp() {
         }
     }
 
+    /**
+     * Set agent version(Described in MANIFEST.MF)
+     */
+    private static void setAgentVersion() {
+        //set default value
+        AGENT_SETTINGS.put("agent.version", "UNKNOWN");
+        try {
+            Enumeration<URL> resources = SnifferConfigInitializer.class.getClassLoader().getResources(JarFile.MANIFEST_NAME);
+            while (resources.hasMoreElements()) {
+                URL url = resources.nextElement();
+                LOGGER.info("SnifferConfigInitializer url:{}", url.toString());
+                try (InputStream is = url.openStream()) {
+                    if (is != null) {
+                        Manifest manifest = new Manifest(is);
+                        Attributes mainAttribs = manifest.getMainAttributes();
+                        String projectName = mainAttribs.getValue("Implementation-Vendor-Id");
+                        if (projectName != null) {
+                            if ("org.apache.skywalking".equals(projectName)) {
+                                String version = mainAttribs.getValue("Implementation-Version");
+                                AGENT_SETTINGS.put("agent.version", version);
```

Comment:
A question, why do you use this kind rather than directly assign the version to `Config.Agent.VERSION`?
AGENT_SETTINGS is used to override default settings.

Reply:
You are right. there is really no need to do this.